app/components/Header/DropdownContents/NotificationsDropdown.js

```diff
@@ -0,0 +1,87 @@
+// @flow
+
+import { Component } from 'react';
+import cx from 'classnames';
+import { activityRenderers } from 'app/components/Feed';
+import Time from 'app/components/Time';
+import styles from './NotificationsDropdown.css';
+import { Link } from 'react-router-dom';
+import ErrorBoundary from 'app/components/ErrorBoundary';
+import { toSpan } from 'app/components/Feed/context';
+
+type Props = {
+  notificationsData: Object,
+  fetchNotifications: () => void,
+  notifications: Array<Object>,
+  markAllNotifications: () => Promise<void>,
+};
+
+type State = {
+  alreadyFetchedNotifications: boolean,
+};
+
+const NotificationElement = ({ notification }: { notification: Object }) => {
+  const renders = activityRenderers[notification.verb];
+  if (renders) {
+    return (
+      <Link to={renders.getURL(notification)}>
+        <div
+          className={cx(
+            styles.notification,
+            !notification.read ? styles.unRead : null
+          )}
+        >
+          <div className={styles.innerNotification}>
+            <div className={styles.icon}>{renders.icon(notification)}</div>
+            <div>
+              {renders.activityHeader(notification, toSpan)}
+              <Time
+                time={notification.updatedAt}
+                wordsAgo
+                style={{ margin: '0', display: 'block' }}
+              />
+            </div>
+          </div>
+        </div>
+      </Link>
+    );
+  }
+  return null;
```

Comment:
Negate this with
```
if (!renders) return;
return (
...
)
```

Reply:
I totally agree, even though it is a direct copy from our code base 🙃

Reply:
I bet a lot of my comments are from copy-pasted code but... 
> When you find a piece of thrash while cleaning out your old apartment do you move it to your new apartment?

Reply:
I know I know, just thinking of the implications here

Reply:
![image](https://user-images.githubusercontent.com/23152018/147279009-f4939466-374c-42bd-b24f-dcc54cabd491.png)